src/V3SchedTiming.cpp

```diff
@@ -115,22 +115,59 @@ AstCCall* TimingKit::createCommit(AstNetlist* const netlistp) {
                 m_commitFuncp->declPrivate(true);
                 scopeTopp->addBlocksp(m_commitFuncp);
             }
-            AstSenTree* const sentreep = activep->sentreep();
-            FileLine* const flp = sentreep->fileline();
-            // Negate the sensitivity. We will commit only if the event wasn't triggered on the
-            // current iteration
-            auto* const negSentreep = sentreep->cloneTree(false);
-            negSentreep->sensesp()->sensp(
-                new AstLogNot{flp, negSentreep->sensesp()->sensp()->unlinkFrBack()});
-            sentreep->addNextHere(negSentreep);
-            auto* const newactp = new AstActive{flp, "", negSentreep};
-            // Create the commit call and put it in the commit function
-            auto* const commitp = new AstCMethodHard{
-                flp, new AstVarRef{flp, schedulerp, VAccess::READWRITE}, VCMethod::SCHED_COMMIT};
-            if (resumep->pinsp()) commitp->addPinsp(resumep->pinsp()->cloneTree(false));
-            commitp->dtypeSetVoid();
-            newactp->addStmtsp(commitp->makeStmt());
-            m_commitFuncp->addStmtsp(newactp);
+
+            // There is a somewhat complicate dance here. If we have a
+            // suspendable process of the form:
+            //    ->evntA;
+            //    $display("Fired evntA");
+            //    @(evntA or evntB);
+            // The firing of the event cannot trigger the event control
+            // following it, as the process is not yet sensitive to the event
+            // when it fires (same applies for change detects). The way the
+            // scheduling works, the @evnt will suspend the process before
+            // the firing of the event is recognized on the next iteration of
+            // the 'act' loop, and hence could incorrectly resume the @evnt
+            // statement. To make this work, whenever a process suspends, it
+            // goes into an "uncommtied" state, so it cannot be resumed
+            // immediately on the next iteration of the 'act' loop, which is
+            // what we want. The question then is, when do we "commit" the
```

Comment:
Prefer you reword to avoid "we".  Some other typos below, run a quick spell check (or perhaps your editor has Emacs flyspell equivalent).

```suggestion
            // goes into an "uncommitted" state, so it cannot be resumed
            // immediately on the next iteration of the 'act' loop, which is
            // what we want. The question then is, when do we "commit" the
```